Flutter-Assembly/food_expiry_tracker
# Implement login and signup auth, app drawer and bottom navigation

## Features
* Login Authentication with google and email/password
* Signup Authentication with google and email/password
* App drawer navigation
* Bottom bar navigation
* Synchronizes active drawer page with active bottom nav item
* Add click listener to ensure all pages are linked to their individual widget screens

## Changes
* Add NavigationProvider
* Update UserProvider
* Add firebase_storage, and cached_network_image to pubspec
* Add page loader while awaiting server request
* Validate login and signup form
* Update user name when user used signup up with email/password option
* Synchronizes active drawer page with active bottom nav item

## How to test feature
* Run app
* Navigate to login or signup page
* Login/Signup with google or email/password
* Click on profile icon or menu icon to open drawer
* You can swap also to open drawer
* Click around to see how pages work

## Screenshots

![image](https://user-images.githubusercontent.com/24432247/98461809-905dec80-21af-11eb-8c09-99e28c7451a4.png)

![image](https://user-images.githubusercontent.com/24432247/98461804-83d99400-21af-11eb-91a2-d36fb0fdb897.png)

![image](https://user-images.githubusercontent.com/24432247/98461818-a23f8f80-21af-11eb-8059-1cb0459df0dc.png)

![image](https://user-images.githubusercontent.com/24432247/98461846-e0d54a00-21af-11eb-81e3-1fc8794c0ca2.png)

![image](https://user-images.githubusercontent.com/24432247/98461926-88eb1300-21b0-11eb-9609-3db8ea473cc3.png)
